services/src/main/java/org/exoplatform/gamification/rest/UserReputationEndpoint.java

```diff
@@ -0,0 +1,291 @@
+
+        gamificationService = CommonsUtils.getService(GamificationService.class);
+
+        identityManager = CommonsUtils.getService(IdentityManager.class);
+
+        badgeService = CommonsUtils.getService(BadgeService.class);
+    }
+
+    @GET
+    @RolesAllowed("users")
+    @Path("point/status")
+    public Response getReputationStatus(@Context UriInfo uriInfo, @Context HttpServletRequest request) {
+
+        ConversationState conversationState = ConversationState.getCurrent();
+
+        if (conversationState != null) {
+            try {
+
+                // Compute user id
+                String actorId = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, conversationState.getIdentity().getUserId(), false).getId();
+
+                JSONObject reputation = new JSONObject();
+                reputation.put("points", gamificationService.getUserGlobalScore(actorId));
+
+
+                return Response.ok().cacheControl(cacheControl).entity(reputation.toString()).build();
+
+            } catch (Exception e) {
+
+                LOG.error("Error listing all rules ", e);
+
+                return Response.serverError()
+                        .cacheControl(cacheControl)
+                        .entity("Error listing all rules")
+                        .build();
+            }
+
+        } else {
+            return Response.status(Response.Status.UNAUTHORIZED)
+                    .cacheControl(cacheControl)
+                    .entity("Unauthorized user")
+                    .build();
+        }
+    }
+
+    @GET
+    @RolesAllowed("users")
+    @Path("badge/all")
+    public Response getUserBadges(@Context UriInfo uriInfo, @Context HttpServletRequest request) {
+
+
+        //String s = Util.getViewerId(uriInfo);
+
+        //Identity current = Utils.getOwnerIdentity();
+        ConversationState conversationState = ConversationState.getCurrent();
+
+        if (conversationState != null) {
+            try {
+                //List<ReputationDTO> allRules = reputationService.getBadgesByUser();
+
+                /** This is a fake */
+                JSONArray allBadges = new JSONArray();
+
+                // Compute user id
+                String actorId = identityManager.getOrCreateIdentity(OrganizationIdentityProvider.NAME, conversationState.getIdentity().getUserId(), false).getId();
+
+                Set<GamificationContextItemEntity> gamificationContextItemEntitySet = gamificationService.getUserGamification(actorId);
+
+                allBadges = buildProfileBadge(gamificationContextItemEntitySet);
+
+                return Response.ok().cacheControl(cacheControl).entity(allBadges.toString()).build();
+
+            } catch (Exception e) {
+
+                LOG.error("Error listing all rules ", e);
+
+                return Response.serverError()
+                        .cacheControl(cacheControl)
+                        .entity("Error listing all rules")
```

Comment:
it's not about listing the rules, the same for the error log before

Reply:
Done